### ISSUE TITLE
warehouse.spec.cy.js started failing with ModuleNotFoundError: No module named 'streamlit.elements.time_widgets'

### ISSUE THREAD

**Issue by vqmarkman** [sundeck-io/OpsCenter#194]
Observed in PR: 

- Fix topn ordering #280
- Overcome the 10k row limit from information_schema.query_history for cost control #282
- Fix ordering and labelling in heatmaps #279

<img width="1263" alt="Screenshot 2023-08-25 at 9 29 18 AM" src="https://github.com/sundeck-io/OpsCenter/assets/104043363/8549a1ac-af05-4734-85ad-5b72b505281f">

Code has been there since day 1.
Runs successfully locally for me:

```
  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  home.spec.cy.js                          00:01        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  labels.spec.cy.js                        01:15        6        2        -        4        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  probes.spec.cy.js                        00:47        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  queries.spec.cy.js                       02:51        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  settings.spec.cy.js                      00:17        4        1        -        3        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  warehouse.spec.cy.js                     00:22        2        1        -        1        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        05:36       17        9        -        8        -  

✨  Done in 403.33s.
```

**Comment by joshelser** [comment]
the dependency version string in pyproject.toml isn't doing what we want it to do. I noticed this when testing Ryan's patch this morning, but found that streamlit=1.16 also doesn't work (so, just "fixing" it wasn't quite right)

**Comment by joshelser** [comment]
```
diff --git a/pyproject.toml b/pyproject.toml
index 200eb15..ef61d27 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ readme = "README.md"
 [tool.poetry.dependencies]
 python = "~3.8"
 plotly = "~5.11"
-streamlit = "^1.16.0"
+streamlit = "~1.24"
 snowflake-snowpark-python = { version = "^1.5.1", extras = ["pandas"] }

 [tool.poetry.group.dev.dependencies]
```

I'm guessing upstream streamlit made a release that got pulled in. this patch changes OpsCenter to be using the version of streamlit I've been using (but isn't matching the version of streamlit from SF's conda repo)

**Comment by vqmarkman** [comment]
@joshelser , thanks for the analysis. It looks like this failure is deterministic and started at particular time. Trying to confirm by collecting all of the PRs that have this failure.

**Comment by vqmarkman** [comment]
failures in the queries.spec.cy.js are the same failures.

**Comment by joshelser** [comment]
> It looks like this failure is deterministic and started at particular time

I'm guessing that streamlit 1.26.0 (or whatever version it's choosing) was just released before the first PR failure but I haven't confirmed this.

**Comment by vqmarkman** [comment]
streamlit is  1.22 now (confirmed in opscenter_testing account)
what worries me is that it does not fail locally for me even if I redeploy with 1.22 ...

**Comment by rymurr** [comment]
what is the status of this bug? I see tests are still failing but its not clear from this bug report why? We are starting to get a rather large backup of PRs blocked on this bug.

**Comment by vqmarkman** [comment]
I could not reproduce this locally (upgraded to 1.22 streamlit and nothing failed for me) and I'm not sure  I understand how to fix it. This is not a test issue, this is product failure to be absolutely clear.

**Comment by rymurr** [comment]
closed with #198
